### PR TITLE
Asset graph handler

### PR DESCRIPTION
**Pull request by jakemac53**
Add a basic handler for the `$graph` path to the dev server which lets you search for nodes and view their inputs/outputs and some basic data about the node.

This could be improved a lot but we can do that as one-offs as needed or as we feel like it.

### CODE REVIEW

File: build_runner/lib/src/generate/watch_impl.dart [hunk 1]
```diff
@@ -297,9 +299,9 @@ class WatchImpl implements BuildState {
 
   /// Checks if we should skip a watch event for this [change].
   bool _shouldProcess(AssetChange change) {
-    assert(_assetGraph != null);
-    if (_isCacheFile(change) && !_assetGraph.contains(change.id)) return false;
-    var node = _assetGraph.get(change.id);
+    assert(assetGraph != null);
```
Comment (natebosch) [inline build_runner/lib/src/generate/watch_impl.dart]: are these changes necessary? should we still refer to `_assetGraph` from within this class?

File: build_runner/lib/src/server/graph_viz.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+window.$build = {}
```
Comment (natebosch) [inline build_runner/lib/src/server/graph_viz.js]: copyright notice on this and other new files

File: build_runner/lib/src/server/graph_viz_main.dart [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import 'dart:convert';
```
Comment (natebosch) [inline build_runner/lib/src/server/graph_viz_main.dart]: why does the file name mention graph viz?

File: build_runner/lib/src/generate/watch_impl.dart (hunk 1 above)
Comment (jakemac53) [reply build_runner/lib/src/generate/watch_impl.dart]: done

File: build_runner/lib/src/server/graph_viz_main.dart (hunk 3 above)
Comment (jakemac53) [reply build_runner/lib/src/server/graph_viz_main.dart]: I just wanted it to be unique, its the entrypoint for the asset graph visualizer

File: build_runner/lib/src/server/graph_viz.js (hunk 2 above)
Comment (jakemac53) [reply build_runner/lib/src/server/graph_viz.js]: done
